selftests/functional/test_output.py

```diff
@@ -254,40 +254,50 @@ def _check_output(path, exps, name):
                              OUTPUT_TEST_CONTENT)
         test.save()
         result = process.run("%s run --job-results-dir %s --disable-sysinfo "
-                             "--test-runner=runner "
-                             "--json - -- %s" % (AVOCADO, self.tmpdir.name, test))
+                             "--json - -- %s" % (AVOCADO,
+                                                 self.tmpdir.name,
+                                                 test))
         res = json.loads(result.stdout_text)
-        joblog = res["debuglog"]
+        joblog = res["tests"][0]["logfile"]
         exps = [b"[stdout] top_print", b"[stdout] top_stdout",
-                b"[stderr] top_stderr", b"[stdout] top_process",
-                b"[stdout] init_print", b"[stdout] init_stdout",
-                b"[stderr] init_stderr", b"[stdout] init_process",
```

Comment:
I'd recommend that, instead of making the `process.run()` output silent with regards to *this* test expectations, we keep checking it, but as a more explicit check.  Example:

```diff
diff --git a/selftests/functional/test_output.py b/selftests/functional/test_output.py
index 9f01268ee..aa24ff5f4 100644
--- a/selftests/functional/test_output.py
+++ b/selftests/functional/test_output.py
@@ -70,7 +70,8 @@ class OutputTest(Test):
         print("init_print")
         sys.stdout.write("init_stdout\\n")
         sys.stderr.write("init_stderr\\n")
-        process.run("/bin/echo init_process")
+        self.log.info("[process.run() stdout] %s",
+                      process.run("/bin/echo init_process").stdout_text)
 
     def test(self):
         print("test_print")
@@ -261,7 +262,7 @@ class OutputTest(TestCaseTmpDir):
         exps = [b"[stdout] top_print", b"[stdout] top_stdout",
                 b"[stderr] top_stderr", b"[stdout] top_process",
                 b"[stdout] init_print", b"[stdout] init_stdout",
-                b"[stderr] init_stderr", b"[stdout] init_process",
+                b"[stderr] init_stderr", b"[process.run() stdout] init_process",
                 b"[stdout] test_print", b"[stdout] test_stdout",
                 b"[stderr] test_stderr", b"[stdout] test_process"]
         _check_output(joblog, exps, "job.log")
```

This would work with both runners, and would provide a way forward for a more explicit API.

Reply:
IIUC, are you also recommending do the same with the method `def test()` and `def __del__()`? I mean: all process.run()' s output, should be inside a self.log?

Reply:
> IIUC, are you also recommending do the same with the method `def test()` and `def __del__()`? I mean: all process.run()' s output, should be inside a self.log?

Correct, that's the proposal.

Reply:
@clebergnu I'm just reading properly this test, I don't think we should expect that log output is going to be part of stdout (as the test name suggests). `self.log()` is not redirecting to stdout. This should be in a different method. So I will make some changes.